### PR TITLE
fix(hooks): add audit trail for auto-retrospective Stop hook (Fixes #2062)

### LINKED ISSUE #2062: Add per-hook audit trail under .agents/.hook-state/auto-retrospective/
Auto-retrospective Stop hook currently writes the retrospective markdown and updates docs/retros/INDEX.md, but does not persist typed hook-state metadata (status/timestamp/outcome) under `.agents/.hook-state/auto-retrospective/` like the other lifecycle hooks do.

CodeRabbit thread PRRT_kwDOQoWRls59lbeE on PR #1763 requested an isolated audit lane so runs are auditable and resumable. Out of scope for #1703 (lifecycle hook infra); track separately to keep PR #1763 focused on the four NEEDS-FIX threads.

**Acceptance**
- Each invoke_auto_retrospective.py run writes a JSON record to `.agents/.hook-state/auto-retrospective/{YYYY-MM-DD}.jsonl` containing: timestamp, status (created|skipped|failed), retro_filename, skip_reason (if any).
- Matches the audit-log pattern used by invoke_false_completion_gate.py (`_write_audit_log`).
- Test added under tests/test_auto_retrospective.py asserting the audit record exists.

Refs PR #1763 thread PRRT_kwDOQoWRls59lbeE.
Refs Issue #1703 (lifecycle hook infrastructure).
Refs ADR-008.

### PR DESCRIPTION
**Pull request by rjmurillo**
Fixes #2062

## Summary
Adds per-hook JSONL audit-trail to invoke_auto_retrospective.py (Stop hook), mirroring the audit pattern in invoke_false_completion_gate.py.

Each Stop hook invocation now appends one record to:
`.agents/.hook-state/auto-retrospective/{YYYY-MM-DD}.jsonl`

Fields: schema, timestamp (ISO-8601 UTC), hook, status (created|skipped|failed), retro_filename, skip_reason.

Mirrored in both Stop hook copies (.claude/hooks and src/copilot-cli/hooks).

## Tests
tests/test_auto_retrospective.py: 8 new cases cover created, three skipped paths (trivial/existing/bypass-env), failed path, edge cases (missing .agents/, unwritable audit dir), and JSONL well-formedness.

Refs PR #1763 (CodeRabbit thread PRRT_kwDOQoWRls59lbeE), Issue #1703, ADR-008.

### CODE REVIEW

File: .claude/hooks/Stop/invoke_auto_retrospective.py [hunk 1]
```diff
@@ -337,6 +341,68 @@ def generate_retrospective(project_dir: Path, today: str) -> Path | None:
...
+            try:
+                f.write(entry + "\n")
+            finally:
+                if _unlock_file is not None:
+                    _unlock_file(f)
+    except OSError as e:
+        print(
+            f"[hook-error] invoke_auto_retrospective audit: {type(e).__name__}: {e}",
+            file=sys.stderr,
+        )
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .claude/hooks/Stop/invoke_auto_retrospective.py]: Fixed in a11f5fa7. `except OSError` is now `except Exception` in `write_audit_log` (both `.claude/hooks/Stop/` and `src/copilot-cli/hooks/sessionEnd/` copies). The Stop hook's fail-open contract now survives unexpected runtime errors (TypeError, ValueError, JSON-serialization edge cases) while still letting BaseException propagate. The divergence from the canonical `invoke_false_completion_gate.write_audit_log` (which uses `OSError`) is documented in the function docstring under "Stricter/looser/different than canonical" per the canonical-source-mirror rule. All 19 tests in tests/test_auto_retrospective.py pass.

File: src/copilot-cli/hooks/sessionEnd/invoke_auto_retrospective.py [hunk 2]
```diff
@@ -337,6 +341,68 @@ def generate_retrospective(project_dir: Path, today: str) -> Path | None:
...
+            try:
+                f.write(entry + "\n")
+            finally:
+                if _unlock_file is not None:
+                    _unlock_file(f)
+    except OSError as e:
+        print(
+            f"[hook-error] invoke_auto_retrospective audit: {type(e).__name__}: {e}",
+            file=sys.stderr,
+        )
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply src/copilot-cli/hooks/sessionEnd/invoke_auto_retrospective.py]: Fixed in a11f5fa7. `except OSError` is now `except Exception` in `write_audit_log` (both `.claude/hooks/Stop/` and `src/copilot-cli/hooks/sessionEnd/` copies). The Stop hook's fail-open contract now survives unexpected runtime errors (TypeError, ValueError, JSON-serialization edge cases) while still letting BaseException propagate. The divergence from the canonical `invoke_false_completion_gate.write_audit_log` (which uses `OSError`) is documented in the function docstring under "Stricter/looser/different than canonical" per the canonical-source-mirror rule. All 19 tests in tests/test_auto_retrospective.py pass.

File: .claude/hooks/Stop/invoke_auto_retrospective.py [hunk 3]
```diff
@@ -337,6 +341,68 @@ def generate_retrospective(project_dir: Path, today: str) -> Path | None:
...
+    project_dir: Path,
+    status: str,
+    retro_filename: str = "",
+    skip_reason: str = "",
+) -> None:
+    """Append a per-hook JSONL audit record for this run.
+
+    Mirrors the audit pattern in ``invoke_false_completion_gate.write_audit_log``
+    so operators can correlate retro creation with completion-gate decisions
+    in the same ``.hook-state/`` tree.
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .claude/hooks/Stop/invoke_auto_retrospective.py]: Fixed in a11f5fa7. The `write_audit_log` docstring now cites the canonical source per `.claude/rules/canonical-source-mirror.md`:

- **Canonical source**: `.claude/hooks/PreToolUse/invoke_false_completion_gate.py` (function `write_audit_log`, lines 191-234), with the load-bearing contract quoted verbatim (audit root, consumer-repo guard, JSONL schema fields, file-handle invariant, fail-open semantics).
- **Documented divergences**: subdirectory layout (`.hook-state/auto-retrospective/` vs canonical flat `.hook-state/`), exception scope (`Exception` vs canonical `OSError`), and payload shape (`status`/`retro_filename`/`skip_reason` vs canonical `command`/`decision`/`reason`/`session_id`/`tool_use_id`).

Applied to both file copies (`.claude/hooks/Stop/` and `src/copilot-cli/hooks/sessionEnd/`), which remain byte-identical.

File: .claude/hooks/Stop/invoke_auto_retrospective.py [hunk 4]
```diff
@@ -428,32 +501,70 @@ def main() -> int:
...
+            project_dir,
+            "failed",
+            skip_reason="generate_retrospective returned None",
+        )
+        return 0
+
+    write_audit_log(
+        project_dir,
+        "created",
+        retro_filename=retro_path.name,
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .claude/hooks/Stop/invoke_auto_retrospective.py:585]: Already fixed in 8400b06e (commit before push a11f5fa7). The retro generation and index update are now in separate try/except blocks:

1. `generate_retrospective` runs in its own try/except. On failure → audit `failed`. On success → audit `created` with `retro_filename=retro_path.name`.
2. `update_retro_index` runs in a separate try/except AFTER the `created` audit is written. If the index update raises, the file-creation audit record is already on disk with the correct filename; the index error is surfaced to stderr without overwriting the `created` status.

This matches the requested "Track the generated path before the index update" approach. Verified in tests/test_auto_retrospective.py (19/19 pass).

File: src/copilot-cli/hooks/sessionEnd/invoke_auto_retrospective.py [hunk 5]
```diff
@@ -428,32 +501,70 @@ def main() -> int:
...
+            project_dir,
+            "failed",
+            skip_reason="generate_retrospective returned None",
+        )
+        return 0
+
+    write_audit_log(
+        project_dir,
+        "created",
+        retro_filename=retro_path.name,
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply src/copilot-cli/hooks/sessionEnd/invoke_auto_retrospective.py:585]: Already fixed in 8400b06e (commit before push a11f5fa7). The retro generation and index update are now in separate try/except blocks:

1. `generate_retrospective` runs in its own try/except. On failure → audit `failed`. On success → audit `created` with `retro_filename=retro_path.name`.
2. `update_retro_index` runs in a separate try/except AFTER the `created` audit is written. If the index update raises, the file-creation audit record is already on disk with the correct filename; the index error is surfaced to stderr without overwriting the `created` status.

This matches the requested "Track the generated path before the index update" approach. Verified in tests/test_auto_retrospective.py (19/19 pass).
